ui/src/components/QuickSetup.tsx

```diff
@@ -203,6 +227,95 @@ const QuickSetup = (props: { onLogin: (credentials?: Credentials) => void }) =>
     }
 }
 
+const PartyRegistry = (props: { parties: PartyDetails[] }) => {
+    const { parties } = props
+
+    const registry = useRegistryLookup()
+
+    const [registryData, setRegistryData] = useState<Map<string, string[]>>(new Map())
+
+    useEffect(() => {
+        if (parties.length > 0) {
+            let partyRegistryMap = new Map<string, string[]>()
+
+            parties.forEach(p => {
+                let roles = []
+
+                if (!!registry.investorMap.get(p.party)) {
+                    roles.push("Investor")
+                }
+                if (!!registry.issuerMap.get(p.party)) {
+                    roles.push("Issuer")
+                }
+                if (!!registry.brokerMap.get(p.party)) {
+                    roles.push("Broker")
+                }
+                if (!!registry.custodianMap.get(p.party)) {
+                    roles.push("Custodian")
+                }
+                if (!!registry.exchangeMap.get(p.party)) {
+                    roles.push("Exchange")
+                }
+                if (!!registry.ccpMap.get(p.party)) {
+                    roles.push("CCP")
+                }
+                if (roles.length > 0) {
+                    partyRegistryMap.set(p.party, roles)
+                }
+            })
+
+            setRegistryData(partyRegistryMap)
+        }
+    }, [registry, parties])
+
+    return (
+        <div className='party-registry-tile'>
+            <p className='login-details dark'>Market Setup</p>
+
+            <Table className='party-registry-table' fixed>
+                <Table.Header>
+                    <Table.HeaderCell>Party</Table.HeaderCell>
+                    <Table.HeaderCell>Role</Table.HeaderCell>
+                </Table.Header>
+                <Table.Body>
+                    {parties.map((p, index) => (
+                        <RegistryTableRow
+                            index={index}
+                            party={p}
+                            roles={registryData.get(p.party) || []}
+                        />
+                    ))}
+                </Table.Body>
+            </Table>
+        </div>
+    )
+}
+
+const RegistryTableRow = (props: { index: number; party: PartyDetails; roles: string[] }) => {
+    const { index, party, roles } = props
+    const oddRow = index % 2 === 0 ? "odd-row" : undefined
+    const partyName = <p className='bold'>{party.partyName}</p>
+
+    if (roles.length === 0) {
+        return (
+            <Table.Row className={oddRow}>
```

Comment:
Looking just at the code, I think there may be a chance that this ends up with `<tr class='undefined'>` in the final page HTML if oddRow is undefined.

It'd be safer to have the ternary on 296 say `"odd-row" : ""` instead, in the case of an even row